eclair-core/src/main/scala/fr/acinq/eclair/io/Client.scala

```diff
@@ -134,6 +134,6 @@ class Client(keyPair: KeyPair, socks5ProxyParams_opt: Option[Socks5ProxyParams],
 
 object Client {
 
-  def props(keyPair: KeyPair, socks5ProxyParams_opt: Option[Socks5ProxyParams], peerConnectionConf: PeerConnection.Conf, switchboard: ActorRef, router: ActorRef, address: InetSocketAddress, remoteNodeId: PublicKey, origin_opt: Option[ActorRef]): Props = Props(new Client(keyPair, socks5ProxyParams_opt, peerConnectionConf, switchboard, router, address, remoteNodeId, origin_opt))
+  def props(keyPair: KeyPair, socks5ProxyParams_opt: Option[Socks5ProxyParams], peerConnectionConf: PeerConnection.Conf, switchboard: ActorRef, router: ActorRef, address: InetSocketAddress, remoteNodeId: PublicKey, origin_opt: Option[typed.ActorRef[PeerConnection.ConnectionResult]]): Props = Props(new Client(keyPair, socks5ProxyParams_opt, peerConnectionConf, switchboard, router, address, remoteNodeId, origin_opt))
```

Comment:
Converting a typed ActorRef to an untyped one is unsafe and you lose all the benefits of using typed actors.

Reply:
It's a trade-off: if we follow your argument, we should potentially update `ActorRef`s across the whole codebase in a single PR, which isn't part of the main feature the PR focuses on and will make it longer to integrate it because there are more unrelated things to review.

Akka introduced these conversions explicitly to make it easy for codebases to update gradually, actor by actor to akka typed.

I guess for this specific PR it doesn't bleed into too many actors in the codebase, so maybe it's ok. @pm47 what is your recommendation as the akka-typed guru?

Reply:
> I don't think you should modify Client, ClientSpawner, Peer and the other existing actors at all to introduce akka typed into them.

It's a tough one: in general I would keep the number of modified files to a minimum. In this case though, since `origin_opt` is used as a callback, using a typed actor as early as possible really brings an improvement. It's true that _akka_ specifically provides `messageAdapters` to handle conversions in typed actors. Those are useful when we can't modify other actors, but a major drawback is that any conversion error or missing adapter will be discovered at runtime. Here I would say that the change in `Client` is trivial enough that it makes it worthwhile.

OTOH, changing parts of the code not directly related to the PR at hand increases the risk of regressions (especially in `ReconnectionTask` which is a tricky one), so we must be careful with that kind of move.

Reply:
Ok interesting, I was a bit afraid of mostly the (seemingly unrelated) changes it required to `Eclair.scala` and `EclairInternalSerializers.scala`, but I'll review it in more details then.

Reply:
I indeed missed those on my first pass. This is not so trivial and shifts the balance towards keeping untyped actors.

I suggest we stick to untyped actors for this PR, and we do a follow up PR that also types `Switchboard`, WDYT @thomash-acinq ?